### DISCUSSION TITLE
Good stuff!

### DISCUSSION THREAD

**Discussion by keithalewis** [cunnane/xloil#1] [General]
Happy to see you are giving the ancient Excel C SDK new legs. It is the key to high performance add-ins.
My thoughts on this are at https://github.com/xlladdins/xll.

**Comment by cunnane** [comment]
Thanks Keith, I often recommend your excellent library - I might have incorporated it into xlOil had I discovered it earlier!  I remain sad the that the C API doesn't get more updates as it's screaming fast; I've tried to reach out to Microsoft to ask about their plans for Excel interop, but I've not heard anything so far.  Would love if the 'async' functions could replace the ancient RTD, i.e. outside the calc cycle, but I think that's a pipedream...

**Comment by keithalewis** [comment]
When I worked at $BIG_BANK they insisted on getting Microsoft to give them a statement about their support for the C SDK. They made nice noises but no promises. The fact is there are many large organizations relying on that and it would be a $BIG_DEAL if they stopped. Don't hold your breath for any updates to it.
RTD solves some hard problems. Its don't-call-me-I'll-call-you protocol is needed to keep Excel performant. The async API probably will not work for active data updating. There's always DDE :-)
Keep up the good work!

**Reply by cunnane** [reply]
Since you can still open a macro sheet with Ctrl-F11 and there's still Lotus 1-2-3 compatibility settings in the options dialog, I'm sure the C API will be with us for some time yet....Microsoft do excel in backwards compatibility!  However, I do feel those async functions were a missed opportunity to replace the ancient RTD tech (which doesn't play nice with array return values and requires COM registration). 
With native async functions, users inevitably hammer the keyboard with impatience, which just resets the calc!

Agree that there's no easy options out there for BIG_CORP to transition to at the moment.
